contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -181,8 +181,9 @@ create_bbf_authid_login_ext(CreateRoleStmt *stmt)
 	/* Advance cmd counter to make the insert visible */
 	CommandCounterIncrement();
 
-	/* Grant membership to guests */
-	if (!role_is_sa(roleid))
+	/* Grant membership of guests */
+	/* sa and fixed server roles except sysadmin should not have membership in database guest roles */
+	if (!(role_is_sa(roleid) || ((get_sysadmin_oid() != roleid) && IS_BBF_FIXED_SERVER_ROLE(stmt->role))))
```

Comment:
1. Why explicit condition for sysadmin? Fixed server roles check should cover it, isn't it?
2. Will need to revoke guest membership from sysadmin from older versions? Because existing code was not blocking guest membership grant to sysadmin it seems.

Reply:
The condition is to exclude sysadmin case because `IS_BBF_FIXED_SERVER_ROLE` will include that and we don't want to exclude membership of guests for sysadmin. Also Sysadmin is admin of the server so no need to remove any priv from it.

Reply:
Lets keep the check either on oid or name for server level role. I think we already have a macro for `IS_ROLENAME_SYSADMIN` so may be convert it to name.

Reply:
We are not additionally getting the oid or the name. Both are already present and I think oid check will be comparatively faster than name, Also we have already used check with oid in the same condition, not getting reason why to make it to `IS_ROLENAME_SYSADMIN `?